src/Components/TagPickerComponent/TagPickerComponent/TagPickerBaseComponent.ts

```diff
@@ -2,244 +2,259 @@ import * as React from 'react';
-import 'whatwg-fetch';
-
-// https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/entity-metadata
-enum EntityMetadataProperties {
-	EntitySetName = "EntitySetName",
-	PrimaryIdAttribute = "PrimaryIdAttribute",
-	PrimaryNameAttribute  = "PrimaryNameAttribute",
-	DisplayCollectionName  = "DisplayCollectionName"
-}
+import { EntityMetadataProperties } from './EntityMetadataProperties'
+import { IWebApi, WebApi } from './WebApi';
 
 export abstract class TagPickerBaseComponent<TInputs, TOutputs> implements ComponentFramework.StandardControl<TInputs, TOutputs> {
+    /**
+     * Parameters passed to the control via CDS.
+     */
+    public relatedEntity: string;
+    public relationshipEntity: string;
+    public relationshipName: string;
+
+    /**
+     * Selected items cache.
+     */
+    public selectedItems: ITag[];
+
+    /**
+     * General PCF properties.
+     */
     private context: ComponentFramework.Context<TInputs>;
-	private notifyOutputChanged: () => void;
+    private notifyOutputChanged: () => void;
     public container: HTMLDivElement;
 
+    /**
+     * WebApi reference
+     */
+    private webAPI: IWebApi;
+
+    /**
+     * Properties related to the current entity.
+     */
     private entityId: string;
     private entityType: string;
 
-	private get tagDisplayName(): string { return this.relatedEntityMetadata ? this.relatedEntityMetadata[EntityMetadataProperties.DisplayCollectionName] : null; }
-	private get idAttribute(): string { return this.relatedEntityMetadata ? this.relatedEntityMetadata[EntityMetadataProperties.PrimaryIdAttribute] : ""; }
-	private get nameAttribute(): string { return this.relatedEntityMetadata ? this.relatedEntityMetadata[EntityMetadataProperties.PrimaryNameAttribute] : ""; }
-
+    /**
+     * Entity metadata.
+     */
     private entityMetadata: ComponentFramework.PropertyHelper.EntityMetadata;
     private relatedEntityMetadata: ComponentFramework.PropertyHelper.EntityMetadata;
 
+    /**
+     * React component properties.
+     */
     private props: ITagPickerProps = {
-		onChange: this.onChange.bind(this),
-		onEmptyInputFocus: this.onEmptyInputFocus.bind(this),
-		onResolveSuggestions: this.onResolveSuggestions.bind(this)
-	}
-
-    public relatedEntity: string;
-    public relationshipEntity: string;
-    public relationshipName: string;
+        onChange: this.onChange.bind(this),
+        onEmptyInputFocus: this.onEmptyInputFocus.bind(this),
+        onResolveSuggestions: this.onResolveSuggestions.bind(this)
+    }
 
-    public selectedItems: ITag[];
+    /**
+     * Helper methods used to access common metadata properties.
+     */
+    private get idAttribute(): string { return this.relatedEntityMetadata ? this.relatedEntityMetadata[EntityMetadataProperties.PrimaryIdAttribute] : ""; }
+    private get nameAttribute(): string { return this.relatedEntityMetadata ? this.relatedEntityMetadata[EntityMetadataProperties.PrimaryNameAttribute] : ""; }
```

Comment:
forgot to mention this in the previous review, but this gives an error/warning saying the target should be es5 or greater. Modify tsconfig accordingly.

Reply:
This appears to be an issue with the tooling as the target is inherited from node_modules/pcf-scripts/tsconfig_base.json which includes a target of es5.